spec/features/ability_spec.rb

```diff
@@ -22,656 +26,636 @@
+
+      visit edit_admin_conference_program_event_type_path(conference2.short_title, conference2.program.event_types.first)
+      expect(current_path).to eq(edit_admin_conference_program_event_type_path(conference2.short_title, conference2.program.event_types.first))
+
+      visit admin_conference_program_difficulty_levels_path(conference2.short_title)
+      expect(current_path).to eq(admin_conference_program_difficulty_levels_path(conference2.short_title))
+
+      visit new_admin_conference_program_difficulty_level_path(conference2.short_title)
+      expect(current_path).to eq(new_admin_conference_program_difficulty_level_path(conference2.short_title))
+
+      visit edit_admin_conference_program_difficulty_level_path(conference2.short_title, conference2.program.difficulty_levels.first)
+      expect(current_path).to eq(edit_admin_conference_program_difficulty_level_path(conference2.short_title, conference2.program.difficulty_levels.first))
+
+      visit admin_conference_schedules_path(conference2.short_title)
+      expect(current_path).to eq(admin_conference_schedules_path(conference2.short_title))
+
+      create(:schedule, program: conference2.program)
+      visit admin_conference_schedule_path(conference2.short_title, conference2.program.schedules.first)
+      expect(current_path).to eq(admin_conference_schedule_path(conference2.short_title, conference2.program.schedules.first))
+
+      visit admin_conference_program_reports_path(conference2.short_title)
+      expect(current_path).to eq(admin_conference_program_reports_path(conference2.short_title))
+
+      visit admin_conference_registrations_path(conference2.short_title)
+      expect(current_path).to eq(admin_conference_registrations_path(conference2.short_title))
+
+      create(:registration, user: create(:user), conference: conference2)
+      visit edit_admin_conference_registration_path(conference2.short_title, conference2.registrations.first)
+      expect(current_path).to eq(root_path)
+
+      visit new_admin_conference_registration_period_path(conference2.short_title)
+      expect(current_path).to eq(root_path)
+
+      create(:registration_period, conference: conference2)
+      visit edit_admin_conference_registration_period_path(conference2.short_title)
+      expect(current_path).to eq(root_path)
+
+      visit admin_conference_questions_path(conference2.short_title)
+      expect(current_path).to eq(root_path)
+
+      visit admin_conference_program_tracks_path(conference2.short_title)
+      expect(current_path).to eq(admin_conference_program_tracks_path(conference2.short_title))
+
+      visit admin_conference_roles_path(conference2.short_title)
+      expect(current_path).to eq(admin_conference_roles_path(conference2.short_title))
+
+      visit admin_conference_emails_path(conference2.short_title)
+      expect(current_path).to eq(admin_conference_emails_path(conference2.short_title))
+    end
 
-    visit admin_confer
```

Comment:
Almost 700 lines!!! 😱 
How about we split this file into different files, one for each role?